python/rmm/_lib/device_buffer.pxd

```diff
@@ -13,13 +13,19 @@
 # limitations under the License.
 
 from libc.stdint cimport uintptr_t
-from libcpp.memory cimport unique_ptr
+from libcpp.memory cimport shared_ptr, unique_ptr
 
 from rmm._cuda.stream cimport Stream
 from rmm._lib.cuda_stream_view cimport cuda_stream_view
 from rmm._lib.memory_resource cimport DeviceMemoryResource
 
 
+cdef extern from "rmm/mr/device/device_memory_resource.hpp" \
+        namespace "rmm::mr" nogil:
+    cdef cppclass device_memory_resource:
+        pass
+
```

Comment:
We should be able `cimport` this  from `memory_resource.pxd` instead of redefining here (see suggestion above).